lib/Yars/Routes.pm

```diff
@@ -82,6 +82,13 @@ sub _get {
           || _redirect_to_remote_stash( $c, $filename, $md5 )
           || $c->reply->not_found;
     };
+
+    if ($c->req->headers->header('X-Use-X-Accel'))
+    {
+        $c->res->headers->add('X-Accel-Redirect', "$dir/$filename");
```

Comment:
Yeah, I see your point. I'm still opposed to it, but it will allow even bizarre configurations to work.